Signal/src/call/CallService.swift

```diff
@@ -1205,6 +1202,10 @@ protocol CallServiceObserver: class {
     public func handleFailedCall(failedCall: SignalCall?, error: CallError) {
         AssertIsOnMainThread()
 
+        if case .assertionError(let description) = error {
+            assertionFailure(description)
```

Comment:
I wanted to make sure you saw this. 

We've taken a lot of care to have the CallService return to a good state when it encounters errors, but we might inadvertently be ignoring some programming errors. I'm proposing we actually crash when we hit assertion errors in the call service for DEBUG builds.

It's likely that we'll iadvertently hit a couple of debug crashes at first which may require us to reclassify the error (e.g. like I did on line 715.)